### PR TITLE
Adding `audio-to-audio` task.

### PR DESCRIPTION
**Pull request by Narsil**
- Removed `audio-source-separation` (it's not `audio-to-audio` and it is a
bit more general). Also the previous version was flawed as there was no way
to return multiple channels.
- Decision to return base64 blobs in JSON. It's not super shiny but it does
the work and enabled easy ways to add annotations for audio streams
(like the instrument labels for source separation).
- Added GZipMiddleware within common, this help base64 encoded json be 
omewhat ok in terms of size.
- Known caveat. Currently we only send 1 channel to the input of the 
pipeline. At the moment of this commit, no model requires stereo or multi
channel input, so keeping that change for later.
- community framework update will follow.
- Because of dockerized versionning we can safely delete
`audio-source-separation`. Old `api-inference-community` will continue to
support those and the hub too. But if you want to use > 0.0.7 then you will
need to migrate.

### CODE REVIEW

File: api-inference-community/docker_images/common/tests/test_api_audio_to_audio.py [hunk 1]
```diff
@@ -49,11 +51,15 @@ def test_simple(self):
...
+        audio = json.loads(response.content)
 
-        self.assertEqual(len(audio.shape), 1)
-        self.assertGreater(audio.shape[0], 1000)
+        self.assertTrue(isinstance(audio, list))
+        self.assertEqual(set(audio[0].keys()), {"blob", "content-type", "label"})
+
+        data = base64.b64decode(audio[0]["blob"])
+        wavform = ffmpeg_read(data)
+        self.assertGreater(wavform.shape[0], 1000)
```
Comment (osanseviero) [inline api-inference-community/docker_images/common/tests/test_api_audio_to_audio.py:63]: Nit: should we add tests for sampling rate and the annotations?

**Review by osanseviero — COMMENTED** [review]
Looks good. I'm taking a look at the other PR and will stamp this one :)

File: api-inference-community/docker_images/common/tests/test_api_audio_to_audio.py (hunk 1 above)
Comment (Narsil) [reply api-inference-community/docker_images/common/tests/test_api_audio_to_audio.py:63]: We can test the type for sure !

**Review by osanseviero — APPROVED** [review]

File: api-inference-community/docker_images/common/tests/test_api_audio_to_audio.py (hunk 1 above)
Comment (osanseviero) [reply api-inference-community/docker_images/common/tests/test_api_audio_to_audio.py:63]: That would be good idea :)

**Comment by osanseviero** [comment]
This is awesome 🚀 Thank you!

File: api-inference-community/docker_images/common/tests/test_api_audio_to_audio.py (hunk 1 above)
Comment (Narsil) [reply api-inference-community/docker_images/common/tests/test_api_audio_to_audio.py:65]: Done !

**Pull request merged**